src/views/internal/student/class/search/ClassSearch.vue

```diff
@@ -100,10 +100,10 @@ export default {
             });
         },
         goToSigninPage() {
-            this.$router.push({ name: "signin" });
+            this.$router.push({ name: "signin" }).catch(err => {});
```

Comment:
Gostei bastante da refatoração. Ficou muito melhor os nomes.

Apenas não entendi o motivo do bloco catch estar ali se ele não está fazendo nada (aparentemente). Talvez eu esteja errado kkkk

Reply:
Então, às vezes está aparecendo um erro no console: NavigationDuplicated Navigating to current location (“/signin”) is not allowed. Dando uma pesquisada achei essa solução. Diminuiu um pouco, mas às vezes ainda aparece esse erro no console.

Reply:
Entendi. Bom, aparentemente essa solução só esconde o erro... ela não está resolvendo. Acredito que seja melhor investigarmos melhor este erro e resolvê-lo de outra maneira